src/administrator/includes/rbacl.php

```diff
@@ -199,4 +201,65 @@ public static function getRoleByUser($userId, $client = '', $clientContentIid =
 
 		return $roles;
 	}
+
+	/**
+	 * Method to Get roles of users again to selected agency.
+	 *
+	 * @param   integer  $contentId  agency id
+	 * @param   integer  $userId    user id
+	 * @param   integer  $roleId    selected role id
+	 *
+	 * @return 	mixed
+	 *
+	 * @since 1.6
+	 */
+	public function getAuthorizedActions($contentId = null, $userId = null, $roleId = null)
+	{
+		if ($contentId == null)
+		{
+			$input = JFactory::getApplication()->input;
+			$contentId = $input->get('aid', '0', 'INT');
+		}
+
+		if ($userId == null)
+		{
+			$userId = JFactory::getUser()->id;
+		}
+
+		// Get subusers actions mapp
+		$userRoleId = self::getRoleByUser($userId, 'com_multiagency', 0);
+
+		if (empty($userRoleId))
+		{
+			$userRoleId = self::getRoleByUser($userId, 'com_multiagency', $contentId);
+		}
+
+		if (!empty($userRoleId))
+		{
+			$db = JFactory::getDBO();
+			$subInQuery = $db->getQuery(true);
```

Comment:
@sanjivani1812 Move the DB operations in the model

Reply:
@thite-amol - getRoleByUser function having DB operation so I put there. Can please suggest a model name.